microservices/post-service/src/main/java/mini_twitter/post_service/service/PostService.java

```diff
@@ -0,0 +1,183 @@
+    public WebResponseDto<String> deletePost(String postId, String token) {
+        logger.info("Request to delete post ID: {}", postId);
+
+        String userId = userServiceClient.getUserIdFromToken(token);
+        logger.info("User ID retrieved from token: {}", userId);
+
+        Optional<Post> postOptional = postRepository.findById(postId);
+        if (postOptional.isPresent()) {
+            Post post = postOptional.get();
+            if (post.getUserId().equals(userId)) {
+                postRepository.delete(post);
+                logger.info("Post ID: {} deleted successfully by user ID: {}", postId, userId);
+                return WebResponseDto.<String>builder()
+                        .data("Post deleted successfully.")
+                        .build();
+            } else {
+                logger.warn("Unauthorized access attempt to delete post ID: {} by user ID: {}", postId, userId);
+                return WebResponseDto.<String>builder()
+                        .errors("Unauthorized access. You can only delete your own posts.")
+                        .build();
+            }
+        } else {
+            logger.error("Post ID: {} not found or user ID: {} is not authorized to delete this post.", postId, userId);
+            return WebResponseDto.<String>builder()
+                    .errors("Post not found or you are not authorized to delete this post.")
+                    .build();
+        }
+    }
+
+    @Transactional
+    public WebResponseDto<PostDetailResponseDto> getPostDetail(String postId) {
+        logger.info("Fetching post details for postId: {}", postId);
+
+        Optional<Post> postOptional = postRepository.findById(postId);
+        if (postOptional.isPresent()) {
+            Post post = postOptional.get();
+            logger.info("Post found for postId: {}", postId);
+
+            var comments = commentServiceClient.getCommentsByPostId(postId);
+            logger.info("Fetched {} comments for postId: {}", comments.size(), postId);
+
+            int likeCount = likeServiceClient.getLikesByPostId(postId);
+            logger.info("Fetched {} likes for postId: {}", likeCount, postId);
+
+            PostDetailResponseDto postDetailResponse = PostDetailResponseDto.builder()
+                    .id(post.getId())
+                    .userId(post.getUserId())
+                    .content(post.getContent())
+                    .createdAt(post.getCreatedAt())
+                    .comments(comments)
+                    .likes(likeCount)
+                    .build();
+
+            return WebResponseDto.<PostDetailResponseDto>builder()
+                    .data(postDetailResponse)
+                    .build();
+        } else {
+            logger.warn("Post not found for postId: {}", postId);
+            return WebResponseDto.<PostDetailResponseDto>builder()
+                    .errors("Post not found.")
+                    .build();
+        }
+    }
+
+    @Transactional
```

Comment:
Why Fetching data need annotation @Transactional ?

Reply:
Thanks for your feedback! I initially added @Transactional to the data-fetching method thinking it was necessary. I understand now that it’s not needed for read-only operations, so I’ve removed it and added try-catch blocks for error handling instead.